### PR TITLE
Update e2e.EnsureIngress() to check Annotations in addition to Spec

### PR DESCRIPTION
**Pull request by spencerhance**
In addition to fixing the e2e func, the following fixes are included in the addl. commits:
* Fix TestILBStaticIP bug where the annotations were not being applied by using the newly fixed EnsureIngress() func instead.
* Fix a missing negAnnotation in the ILB e2e test.

### CODE REVIEW

File: cmd/e2e-test/ilb_test.go [hunk 1]
```diff
@@ -191,9 +191,7 @@ func TestILBStaticIP(t *testing.T) {
 		var gclb *fuzz.GCLB
 		for i, testIng := range []*v1.Ingress{testIngDisabled, testIngEnabled, testIngDisabled} {
 			t.Run(fmt.Sprintf("Transition-%d", i), func(t *testing.T) {
-				newIng := ing.DeepCopy()
-				newIng.Spec = testIng.Spec
-				ing, err = crud.Patch(ing, newIng)
+				ing, err = crud.Patch(ing, testIng)
```
Comment (aojea) [inline cmd/e2e-test/ilb_test.go:194]: do we need to change this too ?

https://github.com/kubernetes/ingress-gce/blob/8cabaedcf3f758aeb4ebb1df0c5f294d5c2eae23/pkg/e2e/fixtures.go#L279

**Comment by aojea** [comment]
/ok-to-test

should not be better one commit per change?

File: cmd/e2e-test/ilb_test.go (hunk 1 above)
Comment (spencerhance) [reply cmd/e2e-test/ilb_test.go]: Good point!  Just updated the PR.

**Pull request merged**